### PR TITLE
Merge back to develop after hotfix

### PR DESCRIPTION
**Pull request by daineAMD**
Merging master back into develop and resolving merge conflicts.

### CODE REVIEW

File: library/src/include/lapack_device_functions.hpp [hunk 1]
```diff
@@ -854,7 +854,7 @@ ROCSOLVER_KERNEL void gemm_kernel(const rocblas_int m,
 
     // shared mem setup
     extern __shared__ double lmem[];
-    T* a = reinterpret_cast<T*>(lmem);
+    T* a = (T*)lmem;
```
Comment (cgmb) [inline library/src/include/lapack_device_functions.hpp:857]: This was changed to `reinterpret_cast` in #354. Why revert it?

Comment (daineAMD) [reply library/src/include/lapack_device_functions.hpp:857]: My fault - I was looking at https://github.com/ROCmSoftwarePlatform/rocSOLVER/commit/80ed7c6f78245ae66125fc4d2b3afd2a8b2268e9 when fixing that merge conflict, didn't realize it was changed afterwards in #354

**Review by tfalders — APPROVED** [review]
Thanks, Daine! LGTM, as long as the `reinterpret_cast` lines are fixed (2 in lapack_device_functions.hpp and 1 in roclapack_getrf.hpp).

**Review by jzuniga-amd — APPROVED** [review]
LGTM

**Review by cgmb — APPROVED** [review]

**Pull request merged**